modules/data-entry/src/main/frontend/src/questionnaire/SubjectTypeDialog.jsx

```diff
@@ -0,0 +1,204 @@
+
+        // If parent changed we need to move the node
+        if (isEdit && initialParent != parentSubject) {
+          moveSubjectType();
+        } else {
+          onSubmit();
+        }
+    });
+  }
+
+  let moveSubjectType = () => {
+    let formData = new FormData();
+    formData.append(':operation', 'move');
+    formData.append(':dest', !parentSubject.endsWith("/") ? parentSubject + "/" : parentSubject);
+    formData.append(':replace', true);
+
+    fetch(editSubject["@path"], {
+        method: 'POST',
+        body: formData
+    })
+    .then((response) => {
+        if (!response.ok) {
+          setError(response.statusText);
+          return;
+        }
+
+        onSubmit();
+    });
+  }
+
+  let close = () => {
+    setError("");
+    setLabel("");
+    setParentSubject("");
+    setOrder(0);
+    setIsDuplicateLabel(false);
+    onClose();
+  }
+
+  return (
+    <Dialog
+      maxWidth="xs"
+      open={open}
+      onClose={onClose}
+    >
+      <DialogTitle>{isEdit ? "Modify " + editSubject.label : "Create New Subject Type"}</DialogTitle>
+      <DialogContent>
+        <Grid container justify="flex-start" alignItems="center" spacing={2}>
+          <Grid item xs={4}>
+            <Typography>Label</Typography>
+          </Grid>
+          <Grid item xs={8}>
+            <TextField
+                fullWidth
+                value={label}
+                id="label"
+                name="label"
+                onChange={(event) => { setLabel(event.target.value); validateLabel(event.target.value); }}
+                autoFocus
+                error={isDuplicateLabel}
+                helperText={isDuplicateLabel ? "This label already exists" : "Required*"}
+            />
+          </Grid>
+          { (isEdit || subjectTypes && subjectTypes.length > 0) &&
+            <>
+              <Grid item xs={4}>
+                <Typography>Parent</Typography>
+              </Grid>
+              <Grid item xs={8}>
+                <Select
+                  disabled={isEdit && editSubject.subjectsNumber != undefined && editSubject.subjectsNumber > 0}
+                  labelId="parent"
+                  label="optional"
+                  value={parentSubject}
+                  onChange={(event) => { setParentSubject(event.target.value); setError(""); }}
+                  displayEmpty
+                >
+                  <MenuItem key="none" value="/SubjectTypes">
+                    <em>None</em>
+                  </MenuItem>
+                  { subjectTypes.map((option) =>
+                      <MenuItem key={option["jcr:uuid"]} value={option["@path"]}>
+                        {option.label}
+                      </MenuItem>
+                    )
+                  }
+                </Select>
+                <FormHelperText>{isEdit && editSubject.subjectsNumber && editSubject.subjectsNumber > 0 && "There are already subjects of this type. The parent can no longer be changed"}</FormHelperText>
```

Comment:
"0" displayed under the select when there are no subjects of that type:

```suggestion
                { isEdit && editSubject.subjectsNumber > 0 && 
                  <FormHelperText>There are already subjects of this type. The parent can no longer be changed./FormHelperText>
                }
```
Also, the parent for Tumor is still missing:
![image](https://user-images.githubusercontent.com/651980/111822800-992ba100-88ba-11eb-8901-b1485447992b.png)